pandas/tests/dtypes/test_inference.py

```diff
@@ -591,6 +591,19 @@ def test_unicode(self):
         expected = 'unicode' if PY2 else 'string'
         assert result == expected
 
+    @pytest.mark.parametrize('dtype, skipna, expected', [
+        (float, False, 'floating'),
+        (float, True, 'floating'),
+        (object, False, 'floating'),
+        (object, True, 'empty')
+    ])
+    def test_object_empty(self, dtype, skipna, expected):
+        # GH 23421
+        arr = pd.Series([np.nan, np.nan], dtype=dtype)
+
+        result = lib.infer_dtype(arr, skipna=skipna)
```

Comment:
also tests these ndarrays

Reply:
hope I understood this correctly. Parametrizing over `box` in `[pd.Series, np.array]`.